src/main/resources/api/service_openapi.yaml

```diff
@@ -50,6 +81,16 @@ components:
       required: true
       schema:
         type: string
+
+    Type:
+      name: type
+      in: path
+      description: The reousrce type
+      required: true
+      schema:
+        type: string
+        enum: [google-bigquery-dataset, google-bigquery-table, google-blob, google-bucket, google-project]
```

Comment:
I don't think we need this. As long as we use a CloudResourceUid schema object in the request, we can infer the resource type from that.

Reply:
So it would be CloudResourceUid isinstanceof GoogleBucketUID in the code?

Reply:
CloudResourceUid in Java is a super class of GoogleBucketUid. So when we get the CloudResourceUid from the request, we can do a series of if `instanceOf GoogleBucketUid` to cast to the right subclass.